service/resource/legacyv2/resource.go

```diff
@@ -145,6 +145,39 @@ func (r *Resource) newCreateChange(ctx context.Context, obj, currentState, desir
 		return nil, microerror.Mask(err)
 	}
 
+	// Create a service account for the daemonset
+	{
+		serviceAccount := newServiceAccount(customObject)
+		_, err := r.k8sClient.CoreV1().ServiceAccounts(networkNamespace(customObject.Spec)).Create(serviceAccount)
+		if apierrors.IsAlreadyExists(err) {
+			r.logger.Log("debug", "serviceAccount "+serviceAccount.Name+" already exists", "event", "add", "cluster", customObject.Spec.Cluster.ID)
+		} else if err != nil {
+			return nil, microerror.Maskf(err, "creating serviceAccount %s", serviceAccount.Name)
+		}
+	}
+
+	// Bind the service account with the cluster role of flannel operator
+	{
+		clusterRoleBinding := newClusterRoleBinding(customObject)
+		_, err := r.k8sClient.RbacV1beta1().ClusterRoleBindings().Create(clusterRoleBinding)
+		if apierrors.IsAlreadyExists(err) {
+			r.logger.Log("debug", "clusterRoleBinding "+clusterRoleBinding.Name+" already exists", "event", "add", "cluster", customObject.Spec.Cluster.ID)
+		} else if err != nil {
+			return nil, microerror.Maskf(err, "creating clusterRoleBinding %s", clusterRoleBinding.Name)
+		}
+	}
+
+	// Bind the service account with the cluster role of flannel operator pod security policy
+	{
+		clusterRoleBindingPodSecurityPolicy := newClusterRoleBindingPodSecurityPolicy(customObject)
+		_, err := r.k8sClient.RbacV1beta1().ClusterRoleBindings().Create(clusterRoleBindingPodSecurityPolicy)
+		if apierrors.IsAlreadyExists(err) {
+			r.logger.Log("debug", "clusterRoleBindingPodSecurityPolicy "+clusterRoleBindingPodSecurityPolicy.Name+" already exists", "event", "add", "cluster", customObject.Spec.Cluster.ID)
+		} else if err != nil {
+			return nil, microerror.Maskf(err, "creating clusterRoleBindingPodSecurityPolicy %s", clusterRoleBindingPodSecurityPolicy.Name)
+		}
+	}
```

Comment:
Shouldn't those guys also be deleted somewhere?

Reply:
indeed good point